api/orchestration_movers_test.go

```diff
@@ -203,3 +229,121 @@ func Test_stopTaskRun(t *testing.T) {
 		}
 	}
 }
+
+func TestGetRunListById(t *testing.T) {
+
+	run := &DatamoverRun{
+		BytesTransferred:         aws.Int64(100),
+		BytesWritten:             aws.Int64(100),
+		EstimatedBytesToTransfer: aws.Int64(100),
+		EstimatedFilesToTransfer: aws.Int64(100),
+		FilesTransferred:         aws.Int64(100),
+		Status:                   aws.String("RUNNING"),
+		Result:                   &datasync.TaskExecutionResultDetail{},
+		StartTime:                aws.Time(testTime),
+	}
+
+	type input struct {
+		group string
+		name  string
+		id    string
+	}
+	type output struct {
+		isErr bool
+		run   *DatamoverRun
+	}
+	tests := []struct {
+		name  string
+		input input
+		exp   output
+	}{
+		{"group empty", input{"", "name1", "1234"}, output{true, nil}},
+		{"name empty", input{"group1", "", "1234"}, output{true, nil}},
+		{"id empty", input{"group1", "name1", ""}, output{true, nil}},
+		{"valid test", input{"group1", "name1", "1234"}, output{false, run}},
+	}
+
+	o := newMockDataSyncOrchestrator(t)
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ctx := context.Background()
+			got, err := o.datamoverRunDescribe(ctx, tt.input.group, tt.input.name, tt.input.id)
+			if tt.exp.isErr && err == nil {
+				t.Error("expected error but did not receive error")
+			} else if !tt.exp.isErr && err != nil {
+				t.Errorf("received unexpected error %v", err)
+			}
+			assert.Equal(t, tt.exp.run, got)
+		})
+	}
+}
+
+func TestGetRunList(t *testing.T) {
+	validCaseRes := []string{"exec-086d6c629a6bf3581", "exec-086d6c629a6bf3582", "exec-086d6c629a6bf3583", "exec-086d6c629a6bf3584", "exec-086d6c629a6bf3585"}
+	type input struct {
+		group string
+		name  string
+	}
+	type output struct {
+		isErr bool
+		res   []string
+	}
+	tests := []struct {
+		name  string
+		input input
+		exp   output
+	}{
+		{"group empty", input{"", "name1"}, output{true, nil}},
+		{"name empty", input{"group1", ""}, output{true, nil}},
+		{"valid test", input{"group1", "name1"}, output{false, validCaseRes}},
+	}
+
+	o := newMockDataSyncOrchestrator(t)
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ctx := context.Background()
+			got, err := o.datamoverRunList(ctx, tt.input.group, tt.input.name)
+			if tt.exp.isErr && err == nil {
+				t.Error("expected error but did not receive error")
+			} else if !tt.exp.isErr && err != nil {
+				t.Errorf("received unexpected error %v", err)
+			}
+			assert.Equal(t, tt.exp.res, got)
+		})
+	}
+}
+
+func TestGetMoversList(t *testing.T) {
```

Comment:
All fields of the datasyncorchestrator are concrete structs, so we won't be able test all the code paths properly. Replacing those fields with interfaces will allow us to mock them out and properly test all the code paths in orchestrator_movers.go

Reply:
I'm not sure what you mean exactly, but I don't think it's a good idea to use interfaces (unless absolutely necessary).